tests/Tutor.Infrastructure.Tests/TestData/EventStore/EventQueryable/EventQueryableTestCase.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text.Json;
+using Tutor.Core.BuildingBlocks.EventSourcing;
+using Tutor.Infrastructure.Database.EventStore;
+
+namespace Tutor.Infrastructure.Tests.TestData.EventStore
+{
+    public class EventQueryableTestCase
+    {
+        public IEnumerable<QueryParameter> QueryParameters { get; set; } = new List<QueryParameter>();
+
+        public IEventQueryable Execute(IEventQueryable queryable)
+        {
+            var random = new Random(Guid.NewGuid().GetHashCode());
+            var parametersInRandomOrder = QueryParameters.OrderBy(_ => random.Next());
+
+            var result = queryable;
+            foreach (var parameter in parametersInRandomOrder)
+                result = parameter.ApplyToEventQueryable(result);
+            return result;
+        }
+
+        public IEnumerable<DomainEvent> GetExpectedResult(IEnumerable<DomainEvent> events)
+        {
+            var result = events;
+            foreach (var parameter in QueryParameters)
+                result = parameter.ApplyToEnumerable(result);
+            return result;
+        }
+
+        public interface QueryParameter
```

Comment:
IQueryParameter to respect the convention.

Reply:
Corrected in 02acfa3104adbf25dc1aaf41773e57833916d87c.